Zend/tests/type_declarations/intersection_types/invalid_callable_type.phpt

```diff
@@ -0,0 +1,10 @@
+--TEST--
+callable type cannot take part in an intersection type
+--FILE--
+<?php
+
+function foo(): callable&Iterator {}
+
+?>
+--EXPECTF--
+Fatal error: Type callable cannot be part of an intersection type in %s on line %d
```

Comment:
wouldn't make sense to allow `callable&string` for example?

usage example:
```
function get_function_attributes(callable&string $function_name): array { ... }
```

this restricts `$function_name` to be a `callable&string`, meaning passing an invokable object, closure, or an array of `array{0: object, 1: string}` would be disallowed.

---

same applies for `callable&array`, and `callable&object` i suppose.

Reply:
This looks interesting, but is there any practical need in such restrictions?

Reply:
where you want the parameter to be a callable object, not just callable, or a callable array ... etc

probably won't be widely used, but still valid.

Reply:
Ehhhhhhhhhhh, not really a fan of that NGL, especially as I'm thinking of adding proper callable types, and IMHO you shouldn't care what sort of callable it is.
Also I need to do special handling for that and at that point might also accept `self` and `static` at that point which I think is rather useless.